packages/core/src/runtime-engine.ts

```diff
@@ -31,9 +31,11 @@ export class RuntimeEngine {
         for (const feature of features) {
             this.initFeature(feature, envName);
         }
+        const runPromises: Array<Promise<void>> = [];
         for (const feature of features) {
-            this.runFeature(feature, envName);
+            runPromises.push(this.runFeature(feature, envName));
         }
+        await Promise.all(runPromises);
```

Comment:
why do you need to await here can just return the promise

Reply:
didn't want values to leak